packages/-ember-data/tests/helpers/qunit-asserts/assert-warning.ts

```diff
@@ -0,0 +1,168 @@
+import QUnit from 'qunit';
+import { registerWarnHandler } from '@ember/debug';
+import { checkMatcher } from './check-matcher';
+import RSVP from 'rsvp';
+
+let HAS_REGISTERED = false;
+let WARNINGS_FOR_TEST: FoundWarning[];
+let HANDLED_WARNINGS_FOR_TEST: FoundWarning[];
+
+interface WarningConfig {
+  id: string;
+  count?: number;
+  until?: string;
+  message?: string | RegExp;
+  url?: string;
+}
+interface FoundWarning {
+  message: string;
+  options: {
+    id: string;
+    message?: string;
+    until?: string;
+    url?: string;
+  };
+}
+
+interface AssertSomeResult {
+  result: boolean;
+  actual: { id: string; count: number };
+  expected: { id: string; count: number };
+  message: string;
+}
+interface AssertNoneResult {
+  result: boolean;
+  actual: FoundWarning[];
+  expected: FoundWarning[];
+  message: string;
+}
+
+/**
+ * Returns a qunit assert result object which passes if the given warning
+ * `id` was found *exactly* `count` times.
+ *
+ * Fails if not found or found more or less than `count`.
+ * Fails if `until` not specified
+ * Optionally fails if `until` has been passed.
+ */
+function expectWarning(config: WarningConfig): AssertSomeResult {
```

Comment:
Can we tweak the name of this function a bit? I find it fairly odd that this is `expectWarning` **and** the `QUnit.assert` method is `expectWarning` but they are quite different conceptually.

I'd suggest something like `validateWarnings` or `checkWarnings`.